app/services/fact_cleaner.rb

```diff
@@ -51,7 +51,7 @@ def delete_excluded_facts
     logger.debug "Cleaning facts mathing excluded pattern: #{excluded_facts_regex}"
 
     FactName.unscoped.reorder(nil).find_in_batches(:batch_size => @batch_size) do |names_batch|
-      fact_name_ids = names_batch.select! { |fact_name| fact_name.name.match(excluded_facts_regex) }.map(&:id)
+      fact_name_ids = names_batch.select { |fact_name| fact_name.name.match(excluded_facts_regex) }.map(&:id)
```

Comment:
would it make sense to do
```suggestion
      fact_name_ids = names_batch.map { |fact_name| fact_name.id if fact_name.name.match(excluded_facts_regex) }.compact
```
to avoid duplicating the facts arrays in memory?
not sure if it would be a significant gain so not a blocker, just a thought